### PR TITLE
2 chainAction2 settings and cleanup

### PR DESCRIPTION
**Pull request by chrisnojima**

### CODE REVIEW

File: shared/actions/json/settings.json [hunk 1]
```diff
@@ -174,9 +174,6 @@
     "processorProfile": {
       "durationSeconds": "number"
     },
-    "waitingForResponse": {
```
Comment (chrisnojima) [inline shared/actions/json/settings.json]: using the waiting store instead which also lets us remove a bunch of chainGenerators

**Comment by chrisnojima** [comment]
@keybase/react-hackers 
cc: @keybase/y2ksquad

File: shared/actions/settings.tsx [hunk 2]
```diff
@@ -17,251 +16,230 @@ import {isAndroidNewerThanN, pprofDir, version} from '../constants/platform'
...
+
+    return [
+      RouteTreeGen.createNavigateUp(),
+      ...(action.payload.thenSignOut ? [ConfigGen.createLogout()] : []),
+    ]
   } catch (error) {
-    yield Saga.put(SettingsGen.createOnUpdatePasswordError({error}))
-  } finally {
-    yield Saga.put(SettingsGen.createWaitingForResponse({waiting: false}))
+    return SettingsGen.createOnUpdatePasswordError({error})
```
Comment (jakob223) [inline shared/actions/settings.tsx]: Why is returning better than `Saga.put`?

File: shared/actions/settings.tsx [hunk 3]
```diff
@@ -347,22 +317,23 @@ function* refreshNotifications() {
...
     SettingsGen.createNotificationsRefreshed({
-      // @ts-ignore
       notifications: I.Map(notifications),
     })
   )
 }
 
-const dbNuke = () => RPCTypes.ctlDbNukeRpcPromise()
+const dbNuke = async () => {
+  await RPCTypes.ctlDbNukeRpcPromise(undefined, Constants.settingsWaitingKey)
```
Comment (jakob223) [inline shared/actions/settings.tsx]: Why `undefined` and not `{}`?

File: shared/actions/settings.tsx (hunk 2 above)
Comment (chrisnojima) [reply shared/actions/settings.tsx]: this isn't a generator anymore. Saga.put is only for sagas to dispatch

File: shared/actions/settings.tsx (hunk 3 above)
Comment (chrisnojima) [reply shared/actions/settings.tsx]: you don't need to return anything from these functions. if you do, then you need to cover all paths (best `return false`). in this case it doesn't return anything. previously it returned whatever `RPCTypes.ctlDbNukeRpcPromise()` returned which is a little dangerous. we essentially never want to plumb that through

**Pull request merged**